ci/pipelines/pipeline.yml

```diff
@@ -0,0 +1,483 @@
+              params:
+                integration_tool: checkout
+                list_changed_files: true
+                submodules: true
+              trigger: true
+              version: every
+            - in_parallel:
+                steps:
+                - do:
+                  - file: pull-request/.git/resource/head_sha
+                    format: trim
+                    load_var: commit
+                    reveal: true
+                  - file: pull-request/.git/resource/pr
+                    format: trim
+                    load_var: pull_request_key
+                    reveal: true
+                  - put: tests-check
+                    inputs: []
+                  - put: pr-build-check
+                    inputs: []
+          - get: android-build-box
+            tags:
+              - large
+            timeout: 5m
+
+    - task: prepare-env
+      image: android-build-box
+      tags:
+        - large
+      file: pull-request/ci/pipelines/tasks/prepare-env.yml
+      input_mapping:
+        source: pull-request
+      output_mapping:
+        source: pull-request
+
+    - task: run-tests
+      image: android-build-box
+      tags:
+        - large
+      file: pull-request/ci/pipelines/tasks/run-tests.yml
+      input_mapping:
+        source: pull-request
+      output_mapping:
+        source: pull-request
+      on_abort:
+        put: tests-check
+        inputs:
+          - tests-check
+        params:
+          conclusion: cancelled
+      on_error:
+        put: tests-check
+        inputs:
+          - tests-check
+        params:
+          conclusion: action_required
+      on_failure:
+        put: tests-check
+        inputs:
+          - tests-check
+        params:
+          conclusion: failure
+          summary: Review the output within Concourse.
+          title: Tests failed
+    
+    - put: tests-check
+      params:
+        conclusion: success
+        summary: Tests ran successfully
+        title: Tests passed
+      inputs:
+        - tests-check
+
+    - task: upload-pr-build
+      file: pull-request/ci/pipelines/tasks/pr-build.yml
+      image: android-build-box
+      tags:
+        - large
+      on_abort:
+        put: pr-build-check
+        inputs:
+          - pr-build-check
+        params:
+          conclusion: cancelled
+      on_error:
+        put: pr-build-check
+        inputs:
+          - pr-build-check
+        params:
+          conclusion: action_required
+      on_failure:
+        put: pr-build-check
+        inputs:
+          - pr-build-check
+        params:
+          conclusion: failure
+          summary: Review the output within Concourse.
+          title: Pull request build failed
+
+    - put: pr-build-check
+      params:
+        conclusion: success
+        summary: Pull request build created successfully
+        title: PR build created
+      inputs:
+        - pr-build-check
+
+    - put: webhooks
+      inputs: []
```

Comment:
You don't need to run this resource on pull requests.

(It really only needs to run "occasionally" to fix up something that might get out of sync. But usually just once is enough.)